src/library_dylink.js

```diff
@@ -483,17 +483,20 @@ var LibraryDylink = {
 
         // initialize the module
 #if USE_PTHREADS
-        // The main thread does a full __wasm_call_ctors (which includes a call
-        // to emscripten_tls_init), but secondary threads should not call static
-        // constructors in general - emscripten_tls_init is the exception.
-        if (ENVIRONMENT_IS_PTHREAD) {
-          var init = moduleExports['emscripten_tls_init'];
-          assert(init);
+        // Only one thread (currently The main thread) should call
+        // __wasm_call_ctors, but all threads need to call emscripten_tls_init
+        var initTLS = moduleExports['emscripten_tls_init'];
+#if ASSERTIONS
+        assert(initTLS);
+#endif
 #if DYLINK_DEBUG
-          out("adding to tlsInitFunctions: " + init);
+        out("adding to tlsInitFunctions: " + initTLS);
 #endif
-          PThread.tlsInitFunctions.push(init);
-        } else {
+        PThread.tlsInitFunctions.push(initTLS);
+        if (runtimeInitialized) {
+          initTLS();
+        }
```

Comment:
It looks like this pushes and calls initTLS whether or not we are in a pthread, but previously both only happened when not in a pthread. Why is this change correct?

Reply:
Before the main thread would run its TLS inits from the static ctor, but the threads would need to run this explicitly.    The point of this change is to remove this difference and always run init TLS explicitly.